pulp_rpm/app/kickstart/treeinfo.py

```diff
@@ -182,15 +166,88 @@ def create_treeinfo(distribution_tree):
     return f
 
 
-class KickstartData:
+class PulpTreeInfo(TreeInfo):
     """
-    Treat parsed kickstart data.
+    Extend TreeInfo for handling errors.
+
+    """
+
+    def deserialize(self, parser):
+        """
+        Handle errors on deserialize TreeInfo.
+
+        """
+        try:
+            super().deserialize(parser)
+        except Exception:
```

Comment:
Would it make sense to catch a more specific type of Exception?

Reply:
when I tested I got a `TypeError` but I'm not so sure which exceptions `productmd.TreeInfo` could raise. I just thought that I should try to use normally `productmd.TreeInfo` and if raises an error, switch to handling by section.